dashboard/src/components/js/Modal/Modal.js

```diff
@@ -21,6 +22,9 @@ const Modal = ({
   staticBackdrop,
 }) => {
   const onClose = () => onModalClose();
+  // Control when users click outside
+  const ref = useRef(null);
+  useOutsideClick(onClose, [ref], showModal);
```

Comment:
Not sure why we need to create a ref here, I guess it's tricky to document it with a test, but maybe useful link to more info, or just a bit more in the above comment?